test/framework/resources/k8s/resources/node.go

```diff
@@ -44,10 +44,23 @@ func (d *defaultNodeManager) GetNodes(nodeLabelKey string, nodeLabelVal string)
 	}
 	ctx := context.Background()
 	nodeList := v1.NodeList{}
+
 	err := d.k8sClient.List(ctx, &nodeList, client.MatchingLabels{
 		nodeLabelKey: nodeLabelVal,
 	})
-	return nodeList, err
+
+	//Filtering control plane nodes from the list of nodes. kOps creates control plane nodes in the
+	//same subnet as worker nodes. Control plane nodes have the label `node-role.kubernetes.io/control-plane`
+	//defined, which can be used to filter out the control plane nodes
+
```

Comment:
bit nit: we need space between `//` and the text for this to pass stricter `go fmt` checks. Also need to remove empty newline before variable definitions